src/games/strategy/net/DesktopUtilityBrowserLauncher.java

```diff
@@ -79,18 +83,19 @@ public static void openFile(final File file) {
    *        A web address (URL) of a web page (ex: "http://www.google.com/")
    */
   public static void openURL(final String url) {
-    try { // attempt to use Desktop library from JDK 1.6+
-      final Class<?> d = Class.forName("java.awt.Desktop");
-      d.getDeclaredMethod("browse", URI.class)
-          .invoke(d.getDeclaredMethod("getDesktop").invoke(null), URI.create(url));
-      // above code mimicks: java.awt.Desktop.getDesktop().browse()
-    } catch (final Exception ignore) { // library not available or failed
+    boolean failed = true;// TODO check if the failed thing is necessary
+    if (Desktop.isDesktopSupported()) {
+      try {
+        Desktop.getDesktop().browse(URI.create(url));
```

Comment:
This block of code is familiar. Seems like you could extract it to a function that returns a boolean, and then set the value of the `failed` variable with it, ie:  `boolean failed = isDesktopSupport(..)` 
Probably would make the failed var unnecessary at this level even, so `if(isDesktopSupported(..))`